### PR TITLE
Reimplement call graph completion, supports interleaved arguments on recursion

### PR DESCRIPTION
**Pull request by imkiva**
Reference:
- [CallGraph.hs#L155](https://github.com/agda/agda/blob/e3bf58d8b2e95bc0481035756f44ddd9fe19b40d/src/full/Agda/Termination/CallGraph.hs#L155)
- [TermCheck.hs#L296](https://github.com/agda/agda/blob/e3bf58d8b2e95bc0481035756f44ddd9fe19b40d/src/full/Agda/Termination/TermCheck.hs#L296)

### CODE REVIEW

File: base/src/main/java/org/aya/terck/CallGraph.java [hunk 1]
```diff
@@ -20,72 +22,126 @@
  * @see CallMatrix
  */
 public record CallGraph<T, P>(
```
Comment (ice1000) [inline base/src/main/java/org/aya/terck/CallGraph.java:24]: It might be a good idea to move this to `tools`?

Comment (imkiva) [reply base/src/main/java/org/aya/terck/CallGraph.java:24]: Yes this is planned

Comment (imkiva) [reply base/src/main/java/org/aya/terck/CallGraph.java:24]: So other projects (like guest, anqur) can have termination check for free!

**Comment by imkiva** [comment]
bors merge

**Comment by imkiva** [comment]
image

**Comment by imkiva** [comment]
bors merge

**Pull request merged**